### DISCUSSION TITLE
Python Syntax for AddTransform

### DISCUSSION THREAD

**Discussion by KS4VA** [thliebig/openEMS-Project#191] [Q&A Usage] [answered]
Hello!
I’m looking to apply transforms to CSXCAD primitives in my openEMS python script. Who knows the syntax for this? Here is my wrong version …

```
CSX = ContinuousStructure()
plate_size = 300 # width of flat plate
#
plate_metal = CSX.AddMetal( 'plate_1' ) # create a perfect electric conductor (PEC)
plate_metal.AddBox([-1, -plate_size/2, -plate_size], [+1, +plate_size/2, 0])
plate_metal.AddTransform('RotateAxis', 'z', 45)  ### <<< this line is wrong 
```
Thanks in advance :-)

**Comment by KS4VA** [comment]
Looks like I also need

from CSXCAD  import CSTransform 
CST = CSTransform.CSTransform() 

... then something like ...

tr = CST.AddTransform()
tr.CST.AddTransform('RotateAxis', 'z', 30)

However the syntax is still not correct. Any help would be appreciated.

**Comment by VolkerMuehlhaus** [comment, marked as answer]
@GWR2 I never used transformation before, but this guess seems to work:

Use a variable to reference the box, and apply transformation to that box variable. I created a quick testcase that worked, and applied to your code it would look like this:

```
box1 = plate_metal.AddBox([-1, -plate_size/2, -plate_size], [+1, +plate_size/2, 0])
box1.AddTransform('RotateAxis', 'z', 45)
```

That's all I had to do in my test code. I did not need any import statements that you showed in #2.

**Comment by KS4VA** [comment]
Thanks so much @VolkerMuehlhaus  !  My Python syntax is weak. That code does indeed work and create the proper geometry that I can see in AppCSXCAD. I'm grateful, thank you. 

I did get one surprise running the model ... the rotated plate seemed "invisible" to the field ... but that seems to be a meshing problem. When I made the plate thicker than the cell size, things work as expected. 

`plate_metal = CSX.AddMetal('plate_1') # create perfect electric conductor (PEC)
box1=plate_metal.AddBox([-25, -plate_size/2, -plate_size], [+25, +plate_size/2, 0])
box1.AddTransform('RotateAxis', 'y', 45)

plate_metal = CSX.AddMetal('plate_2') # create perfect electric conductor (PEC)
box2=plate_metal.AddBox([-25, -plate_size/2, +plate_size], [+25, +plate_size/2, 0])
box2.AddTransform('RotateAxis', 'y', 0)`

![Plates](https://github.com/thliebig/openEMS-Project/assets/162363685/a2913ce6-c531-4a71-898a-e1229ffbddce)
![thicker plates](https://github.com/thliebig/openEMS-Project/assets/162363685/fed86ca6-4f52-45c1-932f-a83781de0c24)

**Comment by VolkerMuehlhaus** [comment]
> I did get one surprise running the model ... the rotated plate seemed "invisible" to the field ... but that seems to be a meshing problem.

I'm not sure how openEMS handles diagonal metals with a coarse mesh, if this creates multiple isolated fragments or possibly one connected staircase object.
